archaius2-core/src/main/java/com/netflix/archaius/config/PropertiesConfig.java

```diff
@@ -0,0 +1,23 @@
+package com.netflix.archaius.config;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+public class PropertiesConfig extends MapConfig {
+
+    public static PropertiesConfig fromFile(String filename) throws IOException {
```

Comment:
> I could modify the implementation to consider classpath as well as file system.

In most of the APIs I have seen, files vs resources are separate and the user explicitly chooses. So if both are needed I would expect two methods, something like:

* `fromFile`
* `fromResources`

Examples from other libraries, guava has both Resources and Files helpers:

http://docs.guava-libraries.googlecode.com/git-history/release/javadoc/com/google/common/io/Resources.html
http://docs.guava-libraries.googlecode.com/git-history/release/javadoc/com/google/common/io/Files.html

Typesafe config has separate methods on the factory:

http://typesafehub.github.io/config/latest/api/com/typesafe/config/ConfigFactory.html

> What, in your opinion, should be the expected behavior if there are multiple matching files?

The others I have looked at archaiusv1 and [typesafe config](http://typesafehub.github.io/config/latest/api/com/typesafe/config/ConfigFactory.html#parseResources-java.lang.Class-java.lang.String-com.typesafe.config.ConfigParseOptions-) both merge. I'm not sure of the override behavior for archaiusv1, but the typesafe config the first encountered has priority. However, there is no guarantee of the order they will be found in scanning the classpath.

I would probably choose to fail or load all and merge. The main argument for merge is that it seems to be more consistent with other libraries. It is just a helper so the user can always implement their own method to load the properties object if they need custom behavior. I think the most important thing is that the behavior is clearly indicated to the user.